embulk-core/src/main/java/org/embulk/exec/BulkLoader.java

```diff
@@ -531,67 +561,100 @@ private ExecutorPlugin newExecutorPlugin(BulkLoaderTask task)
                 task.getExecConfig().get(PluginType.class, "type", PluginType.LOCAL));
     }
 
-    private ExecutionResult doRun(ConfigSource config)
+    private Map<Reporters.Type, ReporterPlugin> newReporterPlugins(final Map<Reporters.Type, ConfigSource> configs)
+    {
+        final ImmutableMap.Builder<Reporters.Type, ReporterPlugin> plugins = ImmutableMap.builder();
+
+        for (final Reporters.Type type : Reporters.Type.values()) {
+            // if the reporter type doesn't appear in Embulk config, it creates new empty config, by default.
+            final ConfigSource config = configs.get(type);
+            plugins.put(type, newReporterPlugin(config));
+        }
+
+        return Maps.immutableEnumMap(plugins.build());
+    }
+
+    private ReporterPlugin newReporterPlugin(final ConfigSource config)
+    {
+        // if 'type' attribute doesn't appear in the reporter section, it returns 'null' reporter plugin.
+        return Exec.newPlugin(ReporterPlugin.class, config.get(PluginType.class, "type", PluginType.NULL));
```

Comment:
Do we use the `null` reporter by default?

(I just thought we may want to use `stdout` just for `event_log`.)